api/client/commands.go

```diff
@@ -2587,3 +2589,107 @@ func (cli *DockerCli) CmdExec(args ...string) error {
 
 	return nil
 }
+
+type containerStats struct {
+	Name             string
+	CpuPercentage    float64
+	Memory           float64
+	MemoryLimit      float64
+	MemoryPercentage float64
+	NetworkRx        float64
+	NetworkTx        float64
+}
+
+func (cli *DockerCli) CmdStats(args ...string) error {
+	cmd := cli.Subcmd("stats", "CONTAINER", "Stream the stats of a container", true)
+	cmd.Require(flag.Min, 1)
+	utils.ParseFlags(cmd, args, true)
+
+	m := &sync.Mutex{}
+	cStats := map[string]containerStats{}
+	for _, name := range cmd.Args() {
+		go cli.streamStats(name, cStats, m)
+	}
+	w := tabwriter.NewWriter(cli.out, 20, 1, 3, ' ', 0)
+	for _ = range time.Tick(500 * time.Millisecond) {
+		fmt.Fprint(cli.out, "\033[2J")
+		fmt.Fprint(cli.out, "\033[H")
+		fmt.Fprintln(w, "CONTAINER\tCPU %\tMEM USAGE/LIMIT\tMEM %\tNET I/O")
+		m.Lock()
+		ss := sortStatsByName(cStats)
+		m.Unlock()
+		for _, s := range ss {
+			fmt.Fprintf(w, "%s\t%.2f%%\t%s/%s\t%.2f%%\t%s/%s\n",
+				s.Name,
+				s.CpuPercentage,
+				units.BytesSize(s.Memory), units.BytesSize(s.MemoryLimit),
+				s.MemoryPercentage,
+				units.BytesSize(s.NetworkRx), units.BytesSize(s.NetworkTx))
+		}
+		w.Flush()
+	}
+	return nil
+}
+
+func (cli *DockerCli) streamStats(name string, data map[string]containerStats, m *sync.Mutex) error {
+	m.Lock()
+	data[name] = containerStats{
+		Name: name,
+	}
+	m.Unlock()
+
+	stream, _, err := cli.call("GET", "/containers/"+name+"/stats", nil, false)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		stream.Close()
+		m.Lock()
+		delete(data, name)
+		m.Unlock()
+	}()
+	var (
+		previousCpu    uint64
+		previousSystem uint64
+		start          = true
+		dec            = json.NewDecoder(stream)
+	)
+	for {
+		var v *stats.Stats
+		if err := dec.Decode(&v); err != nil {
+			return err
+		}
+		var (
+			memPercent = float64(v.MemoryStats.Usage) / float64(v.MemoryStats.Limit) * 100.0
+			cpuPercent = 0.0
+		)
+		if !start {
+			cpuPercent = calcuateCpuPercent(previousCpu, previousSystem, v)
+		}
+		start = false
+		m.Lock()
+		d := data[name]
+		d.CpuPercentage = cpuPercent
+		d.Memory = float64(v.MemoryStats.Usage)
+		d.MemoryLimit = float64(v.MemoryStats.Limit)
+		d.MemoryPercentage = memPercent
+		d.NetworkRx = float64(v.Network.RxBytes)
+		d.NetworkTx = float64(v.Network.TxBytes)
+		data[name] = d
+		m.Unlock()
+
+		previousCpu = v.CpuStats.CpuUsage.TotalUsage
+		previousSystem = v.CpuStats.SystemUsage
+	}
+	return nil
+}
+
+func calcuateCpuPercent(previousCpu, previousSystem uint64, v *stats.Stats) float64 {
+	cpuPercent := 0.0
+	cpuDelta := float64(v.CpuStats.CpuUsage.TotalUsage) - float64(previousCpu)
+	systemDelta := float64(int(v.CpuStats.SystemUsage)/v.ClockTicks) - float64(int(previousSystem)/v.ClockTicks)
+	if systemDelta > 0.0 {
+		cpuPercent = (cpuDelta / systemDelta) * float64(v.ClockTicks*len(v.CpuStats.CpuUsage.PercpuUsage))
```

Comment:
Seems to me that `v.ClockTicks` is used here to get a value to percentages, so for clarity it should be constant `100` instead.